hcid/src/lib.rs

```diff
@@ -172,15 +172,16 @@ impl HcidEncoding {
         }
     }
 
-    /// decode the data from a base32 string with this instance's configuration
+    /// decode the data from a base32 string with this instance's configuration.  Reed-Solomon can
+    /// correct up to 1/2 its parity size worth of erasures (if no other errors are present).
     pub fn decode(&self, data: &str) -> HcidResult<Vec<u8>> {
         // get our parsed data with erasures
         let (data, erasures) = self.pre_decode(data)?;
 
-        if erasures.len() >= self.config.base_parity_byte_count + self.config.cap_parity_byte_count
-        {
-            // our reed-solomon library makes bad corrections once
-            // erasure count equals parity count.
+        if erasures.len() > ( self.config.base_parity_byte_count + self.config.cap_parity_byte_count ) / 2 {
+            // our reed-solomon library makes bad corrections once erasure count exceeds 1/2 the
+            // parity count (it takes 2 parity symbols to find/correct one error, 1 parity symbol to
+            // correct a known erasure)
```

Comment:
woo, i didn't know it was that bad! ah well... someday version 1 : )

Reply:
That's the limit of R-S encoding; it takes 1 Parity symbol to "locate" and error, and 1 Parity symbol to "correct" an erasure.  That's because of the underlying Galois field arithmetic and solving for a position on the polynomial curve (or something mathematical-sounding like that... :) )  So, as soon as I reach parity/2 erasures -- I know that its going to use up all of the remaining parity/2 symbols just to correct the parity/2 erasures I know I have, with *no* reserve parity to detect any other errors...